ambry-store/src/test/java/com.github.ambry.store/LogSegmentTest.java

```diff
@@ -247,11 +247,11 @@ public void readTest() throws IOException {
         }
       }
 
-      // position + buffer.remaining() > endOffset.
+      // position + buffer.remaining() > sizeInBytes.
       long readOverFlowCount = metrics.overflowReadError.getCount();
       try {
-        segment.readInto(readBuf, writeStartOffset + 1);
-        fail("Should have failed to read because position + buffer.remaining() > endOffset");
+        segment.readInto(readBuf, readBuf.remaining() + 1);
```

Comment:
all invalid offsets (positions have been tested in previous block (lines 236 to 248). So, in this block we check for error scenarios where in position + buffer.remaining() > sizeInBytes(). Coz this is the fix we have made. Anyways, I get your point. Will update the patch.